controllers/webhook/webhook.go

```diff
@@ -77,35 +96,26 @@ func (w *OpenDataHubWebhook) checkDupCreation(ctx context.Context, req admission
 		Kind:    req.Kind.Kind,
 	}
 
-	list := &unstructured.UnstructuredList{}
-	list.SetGroupVersionKind(gvk)
-
-	if err := w.client.List(ctx, list); err != nil {
-		return admission.Errored(http.StatusBadRequest, err)
-	}
-
-	// if len == 1 now creation of #2 is being handled
-	if len(list.Items) > 0 {
-		return admission.Denied(fmt.Sprintf("Only one instance of %s object is allowed", req.Kind.Kind))
-	}
-
-	return admission.Allowed(fmt.Sprintf("%s duplication check passed", req.Kind.Kind))
+	// if count == 1 now creation of #2 is being handled
+	return denyCountGtZero(ctx, w.client, gvk,
+		fmt.Sprintf("Only one instance of %s object is allowed", req.Kind.Kind))
 }
 
 func (w *OpenDataHubWebhook) Handle(ctx context.Context, req admission.Request) admission.Response {
 	var resp admission.Response
 
-	// Handle only Create and Update
-	if req.Operation == admissionv1.Delete || req.Operation == admissionv1.Connect {
-		msg := fmt.Sprintf("ODH skipping %v request", req.Operation)
+	switch req.Operation {
+	case admissionv1.Create:
+		resp = w.checkDupCreation(ctx, req)
+	default:
+		msg := fmt.Sprintf("Webhook skipping %v request", req.Operation)
```

Comment:
i think here is meant to say "no logic check by webhook is applied on <operator>'s request" right?

Reply:
> i think here is meant to say "no logic check by webhook is applied on 's request" right?

Pretty much :)